client/src/app/gateways/vote-decrypt-gateway.service.ts

```diff
@@ -0,0 +1,185 @@
+            } else {
+                model.verified = await verify(signature, raws, this._publicMainKey);
+                console.log(`LOG: verified: `, model.id, model.verified)
+            }
+        }
+        if (couldntVerify.length) {
+            console.error(`Couldn't verify the results of the following poll ids: `, couldntVerify.join(`, `));
+        }
+    }
+
+    private unverify(toUnverify: VoteVerificationData[]): void {
+        toUnverify.forEach(date => {
+            const model = this.repo.getViewModel(date.id);
+            if (model) {
+                model.verified = undefined;
+            }
+        });
+    }
+
+    public async encryptVote(poll: ViewPoll, vote: string): Promise<string> {
+        console.log(`----------[encryptVote]----------`);
+        if (!this._publicMainKey || !poll.crypt_key || !poll.crypt_signature) {
+            throw new Error(`Voting failed: Keys not fully loaded.`);
+        }
+
+        const plaintext = new TextEncoder().encode(vote);
+        const cryptKey = uInt8Enc(poll.crypt_key);
+        const cryptSignature = uInt8Enc(poll.crypt_signature);
+        console.log(`transformed: \n>plaintext: `, plaintext, `\n>cryptKey: `, cryptKey, `\n>cryptSignature: `, cryptSignature, `\npublicKey: `, this._publicMainKey);
+
+        if (!(await verify(cryptSignature, cryptKey, this._publicMainKey))) {
+            throw new Error(`Voting failed: Cryptography keys could not be verified.`);
+        }
+        console.log(`verified`);
+
+        const { ephPublKey, sharedSecret } = await this.generateKeysAndSecret(cryptKey);
+        console.log(`>ephPublKey: `, ephPublKey, `\n>sharedSecret: `, sharedSecret);
+
+        const hkdfKey = await this.generateHkdfKey(sharedSecret)
+        console.log(`hkdfKey: `, hkdfKey);
+
+        const nonce = crypto.getRandomValues(new Uint8Array(this._nonceSize));
+        console.log(`nonce: `, nonce);
+
+        const encrypted = new Uint8Array(await (crypto.subtle.encrypt({name: `AES-GCM`, iv: nonce}, hkdfKey, plaintext) as Promise<ArrayBuffer>));
+        console.log(`encrypted: `, encrypted);
+
+        return this.createPayload(ephPublKey, nonce, encrypted);
+    }
+
+    private async generateKeysAndSecret(cryptKey: Uint8Array): Promise<{ephPublKey: Uint8Array, sharedSecret: Uint8Array}> {
+        const privKey = utils.randomPrivateKey();
+        console.log(`>ephPrivKey: `, privKey);
+        const publKey = curve25519.scalarMultBase(privKey);
+        console.log(`public key generated`);
+
+        const secret = curve25519.scalarMult(privKey, cryptKey);
+        return {ephPublKey: publKey, sharedSecret: secret};
+    }
+
+    private async generateHkdfKey(sharedSecret: Uint8Array): Promise<CryptoKey> {
+        return crypto.subtle.deriveKey(
+            {name: `HKDF`, hash: `SHA-256`, salt: new ArrayBuffer(0), info: new ArrayBuffer(0)},
+            await crypto.subtle.importKey(`raw`, sharedSecret.buffer, `HKDF`, true, [`deriveKey`]),
```

Comment:
Please set the 4th param to false. A suppressed exception says th base for derived keys cannot be extractable.